samples/subsys/canbus/isotp/sample.subsys.canbus.isotp fails to build on mimxrt1170_evk_cm7

After the SDK 2.10.0 update for NXP HAL we now get:

`./scripts/twister -s samples/subsys/canbus/isotp/sample.subsys.canbus.isotp -p mimxrt1170_evk_cm7`

gets:

```
/home/galak/git/zephyr/drivers/can/can_mcux_flexcan.c: In function 'mcux_flexcan_init':
/home/galak/git/zephyr/drivers/can/can_mcux_flexcan.c:727:10: error: passing argument 3 of 'FLEXCAN_TransferCreateHandle' from incompatible pointer type [-Werror=incompatible-pointer-types]
  727 |          mcux_flexcan_transfer_callback, data);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |          |
      |          void (*)(CAN_Type *, flexcan_handle_t *, status_t,  uint32_t,  void *) {aka void (*)(CAN_Type *, struct _flexcan_handle *, int,  unsigned int,  void *)}
In file included from /home/galak/git/zephyr/drivers/can/can_mcux_flexcan.c:15:
/home/galak/git/modules/hal/nxp/mcux/drivers/imx/./fsl_flexcan.h:2034:63: note: expected 'flexcan_transfer_callback_t' {aka 'void (*)(CAN_Type *, struct _flexcan_handle *, int,  long long unsigned int,  void *)'} but argument is of type 'void (*)(CAN_Type *, flexcan_handle_t *, status_t,  uint32_t,  void *)' {aka 'void (*)(CAN_Type *, struct _flexcan_handle *, int,  unsigned int,  void *)'}
 2034 |                                   flexcan_transfer_callback_t callback,
      |   
```

--- a/drivers/can/can_mcux_flexcan.c
+++ b/drivers/can/can_mcux_flexcan.c
@@ -506,7 +506,7 @@ static void mcux_flexcan_detach(const struct device *dev, int filter_id)
 }
 
 static inline void mcux_flexcan_transfer_error_status(const struct device *dev,
-						      uint32_t error)
+						      uint64_t error)
 {
 	const struct mcux_flexcan_config *config = dev->config;
 	struct mcux_flexcan_data *data = dev->data;
@@ -518,23 +518,23 @@ static inline void mcux_flexcan_transfer_error_status(const struct device *dev,
 	struct can_bus_err_cnt err_cnt;
 
 	if (error & CAN_ESR1_FLTCONF(2)) {
-		LOG_DBG("Tx bus off (error 0x%08x)", error);
+		LOG_DBG("Tx bus off (error 0x%08llx)", error);
 		status = CAN_TX_BUS_OFF;
 	} else if ((error & kFLEXCAN_Bit0Error) ||
 		   (error & kFLEXCAN_Bit1Error)) {
-		LOG_DBG("TX arbitration lost (error 0x%08x)", error);
+		LOG_DBG("TX arbitration lost (error 0x%08llx)", error);
 		status = CAN_TX_ARB_LOST;
 	} else if (error & kFLEXCAN_AckError) {
-		LOG_DBG("TX no ACK received (error 0x%08x)", error);
+		LOG_DBG("TX no ACK received (error 0x%08llx)", error);
 		status = CAN_TX_ERR;
 	} else if (error & kFLEXCAN_StuffingError) {
-		LOG_DBG("RX stuffing error (error 0x%08x)", error);
+		LOG_DBG("RX stuffing error (error 0x%08llx)", error);
 	} else if (error & kFLEXCAN_FormError) {
-		LOG_DBG("RX form error (error 0x%08x)", error);
+		LOG_DBG("RX form error (error 0x%08llx)", error);
 	} else if (error & kFLEXCAN_CrcError) {
-		LOG_DBG("RX CRC error (error 0x%08x)", error);
+		LOG_DBG("RX CRC error (error 0x%08llx)", error);
 	} else {
-		LOG_DBG("Unhandled error (error 0x%08x)", error);
+		LOG_DBG("Unhandled error (error 0x%08llx)", error);
 	}
 
 	state = mcux_flexcan_get_state(dev, &err_cnt);
@@ -643,10 +643,7 @@ static inline void mcux_flexcan_transfer_rx_idle(const struct device *dev,
 	}
 }
 
-static void mcux_flexcan_transfer_callback(CAN_Type *base,
-					   flexcan_handle_t *handle,
-					   status_t status, uint32_t result,
-					   void *userData)
+static FLEXCAN_CALLBACK(mcux_flexcan_transfer_callback)
 {
 	struct mcux_flexcan_data *data = (struct mcux_flexcan_data *)userData;
 
@@ -654,21 +651,23 @@ static void mcux_flexcan_transfer_callback(CAN_Type *base,
 	case kStatus_FLEXCAN_UnHandled:
 		__fallthrough;
 	case kStatus_FLEXCAN_ErrorStatus:
-		mcux_flexcan_transfer_error_status(data->dev, result);
+		mcux_flexcan_transfer_error_status(data->dev, (uint64_t)result);
 		break;
 	case kStatus_FLEXCAN_TxSwitchToRx:
 		__fallthrough;
 	case kStatus_FLEXCAN_TxIdle:
-		mcux_flexcan_transfer_tx_idle(data->dev, result);
+		/* The result field is a MB value which is limited to 32bit value */
+		mcux_flexcan_transfer_tx_idle(data->dev, (uint32_t)result);
 		break;
 	case kStatus_FLEXCAN_RxOverflow:
 		__fallthrough;
 	case kStatus_FLEXCAN_RxIdle:
-		mcux_flexcan_transfer_rx_idle(data->dev, result);
+		/* The result field is a MB value which is limited to 32bit value */
+		mcux_flexcan_transfer_rx_idle(data->dev, (uint32_t)result);
 		break;
 	default:
 		LOG_WRN("Unhandled error/status (status 0x%08x, "
-			 "result = 0x%08x", status, result);
+			 "result = 0x%08llx", status, (uint64_t)result);
 	}
 }
 

